llama_index/query_engine/retriever_query_engine.py

```diff
@@ -169,16 +169,7 @@ def _query(self, query_bundle: QueryBundle) -> RESPONSE_TYPE:
         with self.callback_manager.event(
             CBEventType.QUERY, payload={EventPayload.QUERY_STR: query_bundle.query_str}
         ) as query_event:
-            with self.callback_manager.event(
```

Comment:
A retriever query engine can work with ANY retriever (and there are a lot)

With this change, only retrieve events for a vector retriever will be logged

I think the ideal solution here is modifying the base retriever class

Reply:
I was thinking this solution, but I found in ***class*** `RetrieverQueryEngine(BaseQueryEngine)`, `event` callback is implemented in it rather than in `BaseQueryEngine`:
```python
def _query(self, query_bundle: QueryBundle) -> RESPONSE_TYPE:
    """Answer a query."""
    with self.callback_manager.event(
        CBEventType.QUERY, payload={EventPayload.QUERY_STR: query_bundle.query_str}
    ) as query_event:
        ...

        query_event.on_end(payload={EventPayload.RESPONSE: response})

    return response
```
In `BaseQueryEngine`, it called `as_trace`:
```python
  def query(self, str_or_query_bundle: QueryType) -> RESPONSE_TYPE:
      with self.callback_manager.as_trace("query"):
          if isinstance(str_or_query_bundle, str):
              str_or_query_bundle = QueryBundle(str_or_query_bundle)
          return self._query(str_or_query_bundle)
```
To make it consistency, I implemented `event` in instance class instead of base class. 
I thought the reason might be that `event` `payload` need case by case customized, whereas `as_trace` only need `trace_id`

## more to discuss
I didn't find much information about `trace vs event`, my understanding is that:
trace: set trace id for it's underlying `events`, `events` with same trace id will be shown in one subpage in phoenix(still multiple `span`)
event: equivalent to `span` in phoenix,

Reply:
Yea so a trace is a collection of events. A trace is usually some action that can contain many events (like a query)

If a trace is created during an existing trace, it is actually ignored and events are added to the current top-level trace.

I think creating a RETRIEVE event + trace in the retriever base class would probably be nice 🤔